File: pkg/generator/testdata/01.go
package testdata

import (
	types "github.com/drewstone/go2rs/pkg/types"
)

var (
	// Data01 - 01.ts
	Data01 = map[string]types.Type{
		"github.com/drewstone/go2rs/pkg/parser/testdata/success.Embedded": &types.Struct{
			Name: "github.com/drewstone/go2rs/pkg/parser/testdata/success.Embedded",
			Fields: map[string]types.StructField{
				"foo": {
					Optional: true,
					Type:     &types.Number{},
				},
			},
		},
		"github.com/drewstone/go2rs/pkg/parser/testdata/success.Status": &types.String{
			Name: "github.com/drewstone/go2rs/pkg/parser/testdata/success.Status",
			Enum: []string{"Failure", "OK"},
		},
		"github.com/drewstone/go2rs/pkg/parser/testdata/success.Data": &types.Struct{
			Name: "github.com/drewstone/go2rs/pkg/parser/testdata/success.Data",
			Fields: map[string]types.StructField{
				"Time": {
					Type: &types.Date{},
				},
				"Package": {
					Type: &types.Nullable{
						Inner: &types.Struct{
							Fields: map[string]types.StructField{
								"data": {
									Type: &types.Number{},
								},
							},
						},
					},
				},
				"foo": {
					Optional: true,
					Type:     &types.Number{},
				},
				"A": {
					Type: &types.Number{},
				},
				"b": {
					Optional: true,
					Type:     &types.Number{},
				},
				"C": {
					Type: &types.String{},
				},
				"D": {
					Type: &types.Nullable{
						Inner: &types.Number{},
					},
				},
				"EnumArray": {
					Type: &types.Array{
						Inner: &types.String{
							Enum: []string{"a", "b", "c"},
						},
					},
				},
				"Array": {
					Type: &types.Nullable{
						Inner: &types.Array{
							Inner: &types.Number{},
						},
					},
				},
				"Map": {
					Type: &types.Map{
						Key: &types.String{},
						Value: &types.String{
							Name: "github.com/drewstone/go2rs/pkg/parser/testdata/success.Status",
							Enum: []string{"Failure", "OK"},
						},
					},
				},
				"OptionalArray": {
					Type: &types.Array{
						Inner: &types.Nullable{
							Inner: &types.String{},
						},
					},
				},
				"Status": {
					Type: &types.String{
						Name: "github.com/drewstone/go2rs/pkg/parser/testdata/success.Status",
						Enum: []string{"Failure", "OK"},
					},
				},
				"Foo": {
					Optional: true,
					Type: &types.Struct{
						Fields: map[string]types.StructField{
							"V": {
								Type: &types.Number{},
							},
						},
					},
				},
				"U": {
					Type: &types.Struct{
						Fields: map[string]types.StructField{
							"Data": {
								Type: &types.Number{},
							},
						},
					},
				},
			},
		},
	}
)

File: pkg/generator/testdata/01.rs
use serde::{Serialize, Deserialize};
use std::collections::HashMap;
use chrono::{DateTime, Utc};

#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum EnumArrayValues {
	A,
	B,
	C,
}

#[derive(Debug, Clone, Copy, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub enum Status {
	Failure,
	OK,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub struct Data {
	#[serde(rename = "A")]
	pub a: u128,
	#[serde(rename = "Array")]
	pub array: Option<Vec<u128>>,
	#[serde(rename = "C")]
	pub c: String,
	#[serde(rename = "D")]
	pub d: Option<u128>,
	#[serde(rename = "EnumArray")]
	pub enum_array: Vec<EnumArrayValues>,
	#[serde(skip_serializing_if = "Option::is_none")]
	pub Foo: Option<Foo>,
	#[serde(rename = "Map")]
	pub map: HashMap<String, Status>,
	#[serde(rename = "OptionalArray")]
	pub optional_array: Vec<Option<String>>,
	#[serde(rename = "Package")]
	pub package: Option<Package>,
	#[serde(rename = "Status")]
	pub status: Status,
	#[serde(rename = "Time")]
	pub time: DateTime<Utc>,
	#[serde(rename = "U")]
	pub u: U,
	#[serde(skip_serializing_if = "Option::is_none")]
	pub b: Option<u128>,
	#[serde(skip_serializing_if = "Option::is_none")]
	pub foo: Option<u128>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub struct Embedded {
	#[serde(skip_serializing_if = "Option::is_none")]
	pub foo: Option<u128>,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub struct Foo {
	#[serde(rename = "V")]
	pub v: u128,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub struct Package {
	pub data: u128,
}

#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "PascalCase")]
pub struct U {
	#[serde(rename = "Data")]
	pub data: u128,
}

